site/zenodo_rdm/legacy/deserializers/metadata.py

```diff
@@ -0,0 +1,487 @@
+            result["family_name"] = human_name.surnames
+
+        return result
+
+    @post_load(pass_original=True)
+    def load_identifiers(self, result, original, **kwargs):
+        """Loads identifiers."""
+        identifiers = []
+
+        orcid = original.get("orcid")
+        gnd = original.get("gnd")
+
+        if orcid:
+            identifiers.append({"scheme": "orcid", "identifier": orcid})
+        if gnd:
+            identifiers.append({"scheme": "gnd", "identifier": gnd})
+
+        if identifiers:
+            result["identifiers"] = identifiers
+
+        return result
+
+
+class CreatorSchema(Schema):
+    """Creator schema."""
+
+    class Meta:
+        """Meta class."""
+
+        unknown = EXCLUDE
+
+    person_or_org = fields.Nested(PersonSchema)
+    affiliations = fields.List(fields.Dict())
+
+    @pre_load
+    def hook_person_or_org(self, data, **kwargs):
+        """Hooks fields to person_or_org."""
+        data["person_or_org"] = {
+            "name": data.get("name"),
+            "orcid": data.get("orcid"),
+            "gnd": data.get("gnd"),
+        }
+        return data
+
+    @post_load(pass_original=True)
+    def load_affiliations(self, result, original, **kwargs):
+        """Loads affiliation."""
+        affiliation = original.get("affiliation")
+
+        if affiliation:
+            result["affiliations"] = [{"name": affiliation}]
+
+        return result
+
+
+class ReferenceSchema(Schema):
+    """Reference schema."""
+
+    reference = SanitizedUnicode()
+
+
+class MetadataSchema(Schema):
+    """Transform the metadata of a record."""
+
+    class Meta:
+        """Meta attributes for the schema."""
+
+        unknown = EXCLUDE
+
+    # TODO add later
+    # if not data:
+    #     raise ValidationError("Empty metadata is not accepted")
+
+    @pre_load
+    def split_identifiers(self, data, **kwargs):
+        """Splits alternate and related identifiers."""
+        # Some identifiers are alternate identifiers if the relation is "isAlternateIdentifier"
+        input_identifiers = data.get("related_identifiers", [])
+        alternate_identifiers = []
+        related_identifiers = []
+        for identifier in input_identifiers:
+            if identifier.get("relation", "") == "isAlternateIdentifier":
+                alternate_identifiers.append(identifier)
+            else:
+                related_identifiers.append(identifier)
+
+        if related_identifiers:
+            data["related_identifiers"] = related_identifiers
+        if alternate_identifiers:
+            data["alternate_identifiers"] = alternate_identifiers
+
+        return data
+
+    title = SanitizedUnicode()
+    description = SanitizedUnicode()
+    publication_date = SanitizedUnicode(default=date.today().isoformat())
+    resource_type = fields.Str()
+    creators = fields.List(fields.Nested(CreatorSchema))
+    publisher = fields.Constant("Zenodo")  # TODO is it?
```

Comment:
minor: Yes, on legacy we didn't have support for custom Publisher. `imprint_publisher` though could go here (since we don't have it in RDM, and rely on `publisher`)